### ISSUE TITLE
Move to Jackson 2.8 or avoid calling GeneratorBase.getOutputContext()

### ISSUE THREAD

**Issue by olivergierke** [elastic/elasticsearch#20089]
Jackson 2.8 changed the method signature of `GeneratorBase.getOutputContext()` which renders code compiled against previous versions of it unusable on 2.8:

```
Caused by: java.lang.NoSuchMethodError: com.fasterxml.jackson.core.base.GeneratorBase.getOutputContext()Lcom/fasterxml/jackson/core/json/JsonWriteContext;
	at org.elasticsearch.common.xcontent.json.JsonXContentGenerator.writeEndRaw(JsonXContentGenerator.java:327)
	at org.elasticsearch.common.xcontent.json.JsonXContentGenerator.writeRawField(JsonXContentGenerator.java:351)
	at org.elasticsearch.common.xcontent.XContentBuilder.rawField(XContentBuilder.java:909)
```

I've filed an [issue with Jackson](https://github.com/FasterXML/jackson-core/issues/313) already, but doubt they're going to revert this as it will break code compiled against 2.8 then :(. Any chance you can avoid calling this method or maybe even upgrade to Jackson 2.8? This issue prevents Spring Data Elasticsearch from running on Jackson 2.8.

**Comment by tlrx** [comment]
We upgraded Jackson to version 2.8.1 on elasticsearch 5.x and recently for future elasticsearch 2.4. We noticed the change in method signature and I think we still need it in elasticsearch for now (though I'll double check this again eventually). I think your issue will be fixed by the upcoming releases of elasticsearch.

**Comment by clintongormley** [comment]
Fixed in 5.0.0 and 2.4.0

**Comment by olivergierke** [comment]
Wow, fantastic! Thanks a lot guys!
